src/tdympfao.c

```diff
@@ -0,0 +1,1601 @@
+        row   = cells[cell_ids_from_to[icell][0]].global_id;
+        for (icol=0; icol<vertex->num_boundary_cells; icol++) {
+          value = Gmatrix[icell][icol] * pBoundary[icol];
+          ierr = VecSetValue(F, row, -value, ADD_VALUES); CHKERRQ(ierr);
+        }
+      }
+      
+      if (cell_ids_from_to[icell][1]>-1 && cells[cell_ids_from_to[icell][1]].is_local) {
+        row   = cells[cell_ids_from_to[icell][1]].global_id;
+        for (icol=0; icol<vertex->num_boundary_cells; icol++) {
+          value = Gmatrix[icell][icol] * pBoundary[icol];
+          ierr = VecSetValue(F, row, value, ADD_VALUES); CHKERRQ(ierr);
+        }
+      }
+      
+      
+    }
+  }
+
+  ierr = VecAssemblyBegin(F); CHKERRQ(ierr);
+  ierr = VecAssemblyEnd  (F); CHKERRQ(ierr);
+  ierr = MatAssemblyBegin(K,MAT_FINAL_ASSEMBLY); CHKERRQ(ierr);
+  ierr = MatAssemblyEnd  (K,MAT_FINAL_ASSEMBLY); CHKERRQ(ierr);
+
+  PetscFunctionReturn(0);
+
+}
+
+/* -------------------------------------------------------------------------- */
+PetscErrorCode TDyMPFAOComputeSystem(TDy tdy,Mat K,Vec F) {
+
+  DM             dm;
+  TDy_mesh       *mesh;
+  TDy_cell       *cells;
+  PetscInt       icell;
+  PetscInt       row;
+  PetscReal      value;
+  PetscInt       dim;
+  PetscErrorCode ierr;
+
+  PetscFunctionBegin;
+
+  dm       = tdy->dm;
+  mesh     = tdy->mesh;
+  cells    = mesh->cells;
+
+  ierr = MatZeroEntries(K);
+  ierr = MatAssemblyBegin(K,MAT_FINAL_ASSEMBLY); CHKERRQ(ierr);
+  ierr = MatAssemblyEnd  (K,MAT_FINAL_ASSEMBLY); CHKERRQ(ierr);
+
+  ierr = TDyMPFAOComputeSystem_InternalVertices(tdy,K,F); CHKERRQ(ierr);
+  ierr = TDyMPFAOComputeSystem_BoundaryVertices_SharedWithInternalVertices(tdy,K,F); CHKERRQ(ierr);
+  ierr = TDyMPFAOComputeSystem_BoundaryVertices_NotSharedWithInternalVertices(tdy,K,F); CHKERRQ(ierr);
+
+  ierr = DMGetDimension(dm,&dim); CHKERRQ(ierr);
+
+  PetscReal f;
+  if (tdy->forcing || tdy->ops->computeforcing) {
+    for (icell=0; icell<tdy->mesh->num_cells; icell++) {
+      if (cells[icell].is_local) {
+        ierr = (*tdy->ops->computeforcing)(tdy, &(tdy->X[icell*dim]), &f, tdy->forcingctx);CHKERRQ(ierr);
+        value = f * cells[icell].volume;
+        row = cells[icell].global_id;
+        ierr = VecSetValue(F, row, value, ADD_VALUES); CHKERRQ(ierr);
+      }
+    }
+  }
+
+  ierr = VecAssemblyBegin(F); CHKERRQ(ierr);
+  ierr = VecAssemblyEnd  (F); CHKERRQ(ierr);
+  ierr = MatAssemblyBegin(K,MAT_FINAL_ASSEMBLY); CHKERRQ(ierr);
+  ierr = MatAssemblyEnd  (K,MAT_FINAL_ASSEMBLY); CHKERRQ(ierr);
+
+  PetscFunctionReturn(0);
+
+}
+
+/* -------------------------------------------------------------------------- */
+
+PetscErrorCode TDyMPFAORecoverVelocity(TDy tdy, Vec U) {
+
+  DM             dm;
+  TDy_mesh       *mesh;
+  TDy_vertex     *vertices, *vertex;
+  TDy_edge       *edges, *edge;
+  PetscInt       ivertex, icell, isubcell;
+  PetscInt       icol, row, col, vertex_id, iedge;
```

Comment:
same deal, `row` is set inside a conditional, initialize it to something.

Reply:
Fixed via https://github.com/TDycores-Project/TDycore/pull/5/commits/e7441c7e94d399ab7f76cee3617631efa9a1ae3a#diff-d55e8fd8865fbd3ac7d2d195600dcc2fR1266